### ISSUE TITLE
Speed cameras: ignore highway=speed_camera if relation=enforcement exits at the same place

### ISSUE THREAD

**Issue by Neuwessi11** [osmandapp/OsmAnd#14461]
If one speed camera is tagged at the same place both with highway=speed_camera and relation=enforcement, OsmAnd doesn't ignore the highway=speed_camera. This is very annoying because I get a lot of wrong announcements that warn me because of speed control.
That is because highway=speed_camera is for both directions of a street, and the relation is only for one direction.

If there is a relation tagged, we can assume that the highway=speed_camera exists only for software that can not handle the relation.

**Comment by vshcherb** [comment]
Can you please share an example?

**Comment by Neuwessi11** [comment]
> Can you please share an example?

Yes gladly!
Take the device on 60.021655, 14.981629
It takes pictures only going south. But when I yesterday made a simulation going north from 60.01591, 14.98559 to 60.03057, 14.97403
I was warned going north, too.
But at this point I come to another weird thing: In the OSM database I have deleted the highway=maxspeed 3 days ago. Some seconds ago I loaded the newest live updates and was still warned going north.
So either there is a problem with the live updates (that they have no 3 days old updates), or there is a problem with the enforcement-relation.

**Comment by Neuwessi11** [comment]
But coming to the original problem, you can take the device on 59.9900403, 15.018693
This device takes pictures only going north, but when you make a routing going south from 59.99341, 15.01629 to 59.98663, 15.02031, you will be warned, too.

**Comment by vshcherb** [comment]
There is definitely a problem with Live updates cause they don't process relation correct i.e. if you update only point and don't change relation, I suspect it will create a duplicate warning but it needs to be checked.
It's possible that speed_camera is not connected to relation and it also causes duplicate but that would be another issue. So if you can share link to osm objects, it would be nice.

59.99341, 15.01629 - I found 1 relation https://www.openstreetmap.org/relation/10823680#map=18/60.02178/14.98251 but didn't find a camera

**Comment by Neuwessi11** [comment]
> 59.99341, 15.01629 - I found 1 relation https://www.openstreetmap.org/relation/10823680#map=18/60.02178/14.98251 but didn't find a camera

Yes, that was what I meant with "I deleted it ..." I recognised too late, that it was this point I deleted. I am sorry!
But at my second example you will find both - camera and relation, with device here: https://www.openstreetmap.org/node/7288909605#map=19/59.99004/15.01869

> So if you can share link to osm objects, it would be nice.

Do you mean I shall share objects with camera and relation where I get unneccessary warnings?

**Comment by vshcherb** [comment]
Thanks for coordinates we will double check

**Comment by Zirochkabila** [comment]
Your case was reproduced, a bug report was created
https://github.com/osmandapp/OsmAnd/issues/14511

> > 59.99341, 15.01629 - I found 1 relation https://www.openstreetmap.org/relation/10823680#map=18/60.02178/14.98251 but didn't find a camera
> 
> Yes, that was what I meant with "I deleted it ..." I recognised too late, that it was this point I deleted. I am sorry! But at my second example you will find both - camera and relation, with device here: https://www.openstreetmap.org/node/7288909605#map=19/59.99004/15.01869
> 
> > So if you can share link to osm objects, it would be nice.
> 
> Do you mean I shall share objects with camera and relation where I get unneccessary warnings?

**Comment by Neuwessi11** [comment]
Yes, I know, but this is a bug. There should be only a warning in the correct direction.

**Comment by Zirochkabila** [comment]
Thank you, that's right, this is where the North-South speed camera warning goes off by mistake

**Comment by vshcherb** [comment]
As a result of investigation we have 2 relation enforcement
- https://www.openstreetmap.org/relation/10823669
- https://www.openstreetmap.org/relation/10823670
- 
Unfortunately OsmAnd doesn't support yet directions of enforcement if they use the same highway (way) underneath.

**Comment by Neuwessi11** [comment]
Only as a question of understanding: OsmAnd does only support directions of enforcement, if the highway is tagged with oneway=yes, but if it is not such a highway, it is not supported?
That means, at the moment OsmAnd utilizes from such a relation=enforcement only the role=device, but not the role=from or role=to?

**Comment by vshcherb** [comment]
https://github.com/Neuwessi11 yes without one way it's not supported, cause from/to are not used directly the camera is assigned to "from" point on the road

**Comment by chereskata** [comment]
@vshcherb, so this means there is no way to check (for example in render.xml, to draw specific icons) what relation member node has what role (role=device, role=from ...)?
